activerecord/test/cases/autosave_association_test.rb

```diff
@@ -1774,3 +1775,27 @@ def test_after_save_callback_with_autosave
     assert_equal 1, comment.post_comments_count
   end
 end
+
+class TestAutosaveAssociationOnAHasManyAssociationDefinedInSubclassWithAcceptsNestedAttributes < ActiveRecord::TestCase
+  class Agency < Firm
+    has_many :projects, foreign_key: :firm_id
+
+    accepts_nested_attributes_for :projects
+  end
```

Comment:
Not sure, but maybe it would be better to move this class definition to `activerecord/test/models/company.rb`

Reply:
Actually, I'm not sure too. This specific class used in this specific place only. It should be useless in other cases. Especially when `activerecord/test/models/company.rb` full of other classes.

Also maybe i should move this spec to `activerecord/test/cases/nested_attributes_test.rb`?